### PR TITLE
Student portal page for end of project

### PR DESCRIPTION
**Pull request by KavsHub**
Issue: #341 (Basic student portal for end of Semester 2)

*Initial commit (Work in progress)*

Currently accessible by: {HOST_IP}/portal

Pretty much follows the same style we currently have for the student planner except when you click on the program it takes you to the pdf template for the plan in a new tab.

image

image

I'm trying to add pagination to it now so that it doesn't load 100 programs on one page.

Open to feedback and naming changes. (Wasn't sure what we wanted to call the page so it's called 'portal' for now)

I also call the staff pdf function, should make a student version for it? or it is safe doing what I'm currently doing? (I've tested and it works when you're not logged in)

### CODE REVIEW

**Comment by KavsHub** [comment]
I might also group programs by year like the way its currently done on CASS's website

File: cassdegrees/static/js/student/list.js [hunk 1]
```diff
@@ -10,4 +10,4 @@ var options = {
   }
 };
 
-var courseSearch = new List('courses', options);
+var programSearch = new List('programs', options);
```
Comment (j-selby) [inline cassdegrees/static/js/student/list.js]: Newline

File: cassdegrees/templates/student/portal.html [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                            </div>
+                        </div>
+                    </div>
+                </a>
+            {% endfor %}
+        </div>
+    </div>
+    <script src="{% static 'js/vendor/list.js' %}" type="application/javascript"></script>
+    <script src="{% static_no_cache 'js/student/list.js' %}" type="application/javascript"></script>
+{% endblock %}
```
Comment (j-selby) [inline cassdegrees/templates/student/portal.html]: Newline.

**Review by j-selby — CHANGES_REQUESTED** [review]
Looks good, but a few comments:

- Link from/to homepage would be good ("Looking for printable PDFs? Click here" kinda deal)
- Make sure to update the breadcrumbs at the top of the page
- This exposes the fact that non-public plans can have PDFs generated - potential security risk

**Comment by KavsHub** [comment]
Pagination has been added. + other small things that were mentioned and discovered.

Link from homepage
image
Pagination (10 per page)
image

**EDIT** I realised the django pagination broke the fuzzy search... looking into it now

**Review by Daniel-Jang — APPROVED** [review]
Seems nice and simple to me, happy to approve.

**Review by JackSekoranja — CHANGES_REQUESTED** [review]
Hate to be the one to point this out, but part of this PR was to hide the student plan creation (since it was only a prototype). What should happen is the base URL goes to the student portal and the prototype should go under ‘/prototype’ or something similar. The prototype should also not be accessible to students, so make sure to use authentication on it

**Comment by KavsHub** [comment]
Okay cool I'll switch the pages and URLs around.

**Comment by KavsHub** [comment]
Okay, the student portal is now the new index.html and I've added authentication to all the student planner (prototype) pages. - Access by going to /prototype

**Comment by KavsHub** [comment]
Issue of security of unpublished plan pdf viewing raised by @j-selby has been solved.
I've now added a login check if trying to view an unpublished plan.

**Review by JackSekoranja — APPROVED** [review]
Very nice solution, happy for you to merge

**Pull request merged**